src/Snapshot.php

```diff
@@ -401,24 +401,56 @@ private function setupSuperGlobalArrays()
     }
 
     /**
-     * @param  mixed   $variable
+     * @param  mixed $variable
      * @return boolean
-     * @todo   Implement this properly
      */
     private function canBeSerialized($variable)
     {
-        if (!is_object($variable)) {
```

Comment:
Instead of truly catching circular references it might be sufficient to increment a counter in each nested `canBeSerialized()` call and return `true` once it's over, say, 50.